hyper/http20/connection.py

```diff
@@ -740,6 +742,12 @@ def _recv_cb(self, stream_id=0):
                 self.recent_recv_streams.discard(stream_id)
                 return
 
+            # make sure to validate the stream is readable.
+            # if the connection was reset, this stream id won't appear in
+            # self.streams and will cause this call to raise an exception.
+            if stream_id:
```

Comment:
Stream IDs *are* sequential. This is part of why I was wondering whether it is really acceptable at all to re-use a connection object in this manner.